### PR TITLE
[DISCO-2515] Contile Centralized Test Docs

### PR DESCRIPTION
**Pull request by taddes**
## References

JIRA: [DISCO-2515](https://mozilla-hub.atlassian.net/browse/DISCO-2515)

## Description
Test documentation centralized in one single location in our `docs/` dir. All pre-existing `README` files are updated with pointers to relevant documentation. This ensures developers working in any repo can easily find and access relevant docs.

All directories and links updated and verbiage changed to reflect new location. 

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/contile/blob/main/CONTRIBUTING.md)
- [x] This PR conforms to the [Opsec policies](https://github.com/mozilla-services/websec-check)
- [ ] [Functional and performance test](https://github.com/mozilla-services/contile/tree/main/test-engineering) coverage has been expanded and maintained (if applicable)
- [ ] The `[load test: (abort|warn)]` keywords are applied (if applicable)
- [x] Documentation has been updated
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title

[DISCO-2515]: https://mozilla-hub.atlassian.net/browse/DISCO-2515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/testing/contract-tests/index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Contract Tests
+
+This is where all contract test related documentation is stored.
+
+- [Overview](./testing/contract-tests/contract-tests.md)
```
Comment (mtbottle) [inline docs/testing/contract-tests/index.md:5]: Similar comment to https://github.com/mozilla-services/merino-py/pull/421#discussion_r1327281765 , I'm wondering if we can reduce one level by putting the overview in this document instead.

File: docs/testing/contract-tests/index.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- [Client](./testing/contract-tests/client.md)
+- [Partner](./testing/contract-tests/partner.md)
```
Comment (mtbottle) [inline docs/testing/contract-tests/index.md:7]: Links are broken.

File: docs/testing/contract-tests/contract-tests.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    * [ ] [docker-compose.init_error.yml][20]
+    * [ ] [docker-compose.tiles_cache.yml][21]
+3. [CircleCI][22] contract test jobs
+    * [ ] [config.yml][23]
+4. Documentation
+    * [ ] contract tests docs [contract-tests.md][24]
+    * [ ] client docs[client.md][4]
+    * [ ] partner docs [partner.md][5]
+
+[0]: ../../../test-engineering/contract/
```
Comment (mtbottle) [inline docs/testing/contract-tests/contract-tests.md:168]: This link is broken, and so are some other ones below.

**Review by mtbottle — CHANGES_REQUESTED** [review]
Generally LGTM. I would suggest going back and checking that all the links work when you create the `mdbook` preview.

File: docs/testing/contract-tests/contract-tests.md (hunk 3 above)
Comment (taddes) [reply docs/testing/contract-tests/contract-tests.md:168]: Hrm, both work in pycharm and VSCode. Is it that mdbook does not support absolute paths, as both IDEs resolve the path? I can change those if that's the case.

**Comment by taddes** [comment]
Links changed to relative links for mdbook docs and all originally internal links to files now point to the GitHub repo files 📖 Also tacked on the other pre-existing broken links in some of the other documentation outside `testing`

**Review by mtbottle — APPROVED** [review]

**Pull request merged**